sphinx_needs/needsfile.py

```diff
@@ -131,6 +131,9 @@ def load_json(self, file: str) -> None:
 
             self.log.debug(f"needs.json file loaded: {file}")
 
+    def add_lut_need(self, version: str, needs_lut: Dict[str, Any]) -> None:
```

Comment:
My idea was a little different.
Instead of adding a custom data structure and use the same writer-function, we could keep the data structure with `add_need()` and just do the output specific formating in a new writer function like `write_lut_json`.
In this case all the code is the same, till it comes to write it to the harddisk.

Reply:
I wonder if we keep formatting JSON and only map need["docname"] = need["external_url"] if need["is_external"] = True, It is easy to reuse flow generate need_lut.json like need.json. Otherwise, I will continue change formatting in *write_lut_json* like {[id] : [docname] or [external_url]}
@r-o-b-e-r-t-o @danwos

Reply:
@nhatnamnguyengtvthcm  Not sure if I got your point. But I think we must have a mapping for internal **and** external needs because otherwise you would bot be redirected correctly by permalink.html in case an external need is referenced.